src/components/FeatureChecking/FeatureFlag.js

```diff
@@ -0,0 +1,30 @@
+// @flow
+import * as React from 'react';
+import { isFeatureEnabled, getFeatureConfig } from './util';
+import FeatureConsumer from './FeatureConsumer';
+import * as types from './flowTypes';
+
+function FeatureFlag({
+    feature,
+    enabled = () => null,
+    disabled = () => null,
+    children,
+}: {
+    feature: string,
+    enabled?: types.FeatureOptions => React.Node,
+    disabled?: () => React.Node,
+    children?: React.Node,
+}) {
+    return (
+        <FeatureConsumer>
+            {features => {
+                const isEnabled = isFeatureEnabled(features, feature);
+                const featureConfig = getFeatureConfig(features, feature);
+                if (children) return isEnabled && children;
+                return isEnabled ? enabled(featureConfig) : disabled();
```

Comment:
What is the purpose of the `enabled` and `disabled` callbacks?

Reply:
allows you to inline logic that depends on the variables in the config object without erroring out

Reply:
It also seems to allow arbitrary components to be rendered as the return value of this function, which I believe we wanted to avoid.

Reply:
No, that's the intended behavior:

```js
function BigThing() {
  return(
    <div>
      <h2>Lots going on in here</h2>
      <FeatureFlag feature="feat-a" enabled={() => <Button>Do Something</Button>} />
      <p>Some other stuff...</p>
      <FeatureFlag feature="feat-b" enabled={() =>
        <div>Other Feature Here</div>
      } />
    </div>
  )
}
```

Reply:
I see. That should be fine as long as the functions passed to `enabled` and `disabled` live inside box-ui-elements and aren't supplied as props by the consuming app.

Reply:
Right, they are props of the internal FeatureFlag component itself